packages/experimental/surface/src/plugins/ListViewPlugin.tsx

```diff
@@ -0,0 +1,151 @@
+//
+// Copyright 2023 DXOS.org
+//
+
+import React, { UIEvent, FC, createContext, useContext, useEffect, useState } from 'react';
+import { useNavigate, useParams } from 'react-router';
+
+import { MulticastObservable } from '@dxos/async';
+import { useMulticastObservable } from '@dxos/react-async';
+
+import { definePlugin, usePluginContext, Plugin } from '../framework';
+
+export type MaybePromise<T> = T | Promise<T>;
+
+export type ListViewItem = {
+  id: string;
+  text: string;
+};
+
+export type ListViewProps = {
+  items?: ListViewItem[];
+  onSelect?: (id: string) => void;
+};
+
+export const ListView = (props: ListViewProps) => {
+  const { items } = props;
+  return (
+    <div>
+      {items?.length
+        ? items.map((item) => (
+            <div key={item.id} onClick={() => props.onSelect?.(item.id)}>
+              {item.text}
+            </div>
+          ))
+        : 'no items'}
+    </div>
+  );
+};
+
+type GraphNode<TDatum = any> = {
+  id: string;
+  label: string;
+  description?: string;
+  icon?: FC;
+  data?: TDatum;
+  actions?: GraphNodeAction[];
+  children?: GraphNode[];
+  parent?: GraphNode;
+};
+
+type GraphNodeAction = {
+  id: string;
+  label: string;
+  icon?: FC;
+  invoke: (event: UIEvent) => MaybePromise<void>;
+};
+
+export type GraphPluginProvides = {
+  graph: {
+    nodes?: (plugins: Plugin[], parent?: GraphNode) => MulticastObservable<GraphNode[]>;
+    actions?: (plugins: Plugin[], parent?: GraphNode) => MulticastObservable<GraphNodeAction[]>;
```

Comment:
actions are labelled event sinks? i.e., "things that can be done to a particular node?"